### PR TITLE
Refactor/repo-manager-as-root

### PR DESCRIPTION
**Pull request by charley-geoffroy**
## What does this PR do ?

This pull request aims to refactor the git-controller project by making the following change:

- Renaming the project from "git-controller" to "git-repo-manager" throughout the entire directory structure, starting from the root folder.

This pull request primarily aims to move the authentication layer and the file system virtualization layer (SSH key FUSE) from this repository to another repository called "git-ssh-server."

By separating these specific layers into the "git-ssh-server" repository, it allows for better modularization and organization of the codebase. This approach simplifies maintenance and future development by creating a dedicated repository focused on handling SSH authentication and handling database requests..

### How should this be manually tested ?

>**_Notes_**: Make sure to have your `.env` file set as needed it

To run the server, run the following command:

```bash
cargo run --bin repo-manager
```

To run the client-debug, run the following command:

```bash
cargo run --bin client-debug
```

### Other changes

- Modifying the `.toml` file to add two new targets: "client-debug" and "repo-manager". This change ensures that these targets are properly included and can be utilized in the project.

- Updating the documentation to reflect the refactored project structure.

### CODE REVIEW

File: .gitignore
Comment (sylvain-pierrot) [inline .gitignore:1]: it would be nice to use the official rust gitingore

File: README.md [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
-hooks, authentification, creating and deleting repositories, etc.
+## Table of Contents
+  * [Description](#description)
+  * [Requirements](#requirements)
+  * [Dependencies](#dependencies)
+  * [Usage](#usage)
+## Description
 
-## Good practices
+Git repo manager is a service to provide a simple way to manage git repositories on a machine. The communication between the client and the server is done via a GRPC Protocol. Here is the server implementation.
```
Comment (sylvain-pierrot) [inline README.md]: It would be good to explain more technically what has been implemented.

**Review by sylvain-pierrot — COMMENTED** [review]
Why is this refactor technically necessary?
Is there a change in approach to the project?

**Review by mmoreiradj — APPROVED** [review]

File: .gitignore
Comment (charley-geoffroy) [reply .gitignore:1]: Done :+1:

**Review by sylvain-pierrot — APPROVED** [review]

**Pull request merged**